pageserver/src/layered_repository.rs

```diff
@@ -102,6 +103,22 @@ lazy_static! {
     .expect("failed to define a metric");
 }
 
+/// The name of the metadata file pageserver creates per timeline.
+pub const METADATA_FILE_NAME: &str = "metadata";
+
+/// All files and data, relevant to a certain pageserver timeline,
+/// found locally in its working directory.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct LocalTimeline {
```

Comment:
The main reason to introduce this is to enforce storing images (aka certain layers in bulk), instead of manipulating disjoint layers.

Also, instead of operating file paths directly, we toss layer names and ids which is enough to construct us a file path in the moment we decide to upload/download the layer.

https://github.com/zenithdb/zenith/pull/686/files#diff-8ef63341d08bdfccc129283af160e1b6032defcd0bc929974510710cdf3d6478R57

Reply:
I think we should introduce a new `LayerFileName` enum that can hold ImageFilename or DeltaFileName. The S3 code shouldn't need to distinguish between different kinds of layers.

The metadata path is always `tenants/{tenant_id}/timelines/{timeline_id}/metadata` , so you don't necessarily need a field for that, you can construct it from the tenant and timeline ids.